### PR TITLE
Mailer

### PR DESCRIPTION
**Pull request by magnus-thor**
### https://www.pivotaltracker.com/story/show/152915383 ###
------------------------------------
- Setup mailer to send order information to kitchen
-------------------------
To test this you need to:  (do not set the gem in gemfile)
- gem install mailcatcher
- mailcatcher
- Go to http://localhost:1080/

screenshot from 2017-11-16 18-20-42

### CODE REVIEW

File: spec/mailers/order_mailer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require "rails_helper"
+
+RSpec.describe OrderMailer, type: :mailer do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (tochman) [inline spec/mailers/order_mailer_spec.rb]: I would like to see some specs for this

File: spec/mailers/previews/order_mailer_preview.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Preview all emails at http://localhost:3000/rails/mailers/order_mailer
+class OrderMailerPreview < ActionMailer::Preview
+
```
Comment (tochman) [inline spec/mailers/previews/order_mailer_preview.rb:3]: What is the responsibility of this class?

File: app/views/order_mailer/welcome_email.text.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+To login to the site, just follow this link: <%= @url %>.
+
+Thanks for joining and have a great day!
+
+
+<!--What we need-->
+<!--customer email-->
+<!--time of the order-->
+<!--order details with dish name, ingredient and quantity-->
```
Comment (tochman) [inline app/views/order_mailer/welcome_email.text.erb]: Not sure if this one is needed

File: app/views/order_mailer/order.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+<head>
+  <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+</head>
+<body>
+<h1> Customer: <%= @customer_email  %>, Pickup time: <%= @pickup_time  %></h1>
+<% @cart.each do |item| %>
+  <h2><%=  item[:product_name] %> <%= item[:ingredient_name] unless item[:ingredient_name].nil? %> x<%= item[:item_quantity] %></h2>
+<% end %>
+</body>
+</html>
```
Comment (tochman) [inline app/views/order_mailer/order.html.erb]: Good for first iteration

File: app/mailers/order_mailer.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class OrderMailer < ApplicationMailer
+
+  def order(customer_email, cart, pickup_time)
+    @customer_email = customer_email
+    @cart = cart
+    @pickup_time = pickup_time
+    @kitchen_email = 'someone.nowhere@mail.com' # https://navigator-lxa.mail.com
+    @url  = 'https://kimchistan-dev.surge.sh/'  #needs  to be changed before production to real address
+    mail to: @kitchen_email, subject: "Pickup time: #{@pickup_time}"
```
Comment (tochman) [inline app/mailers/order_mailer.rb]: Some of the variables can be set to local

File: app/mailers/order_mailer.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class OrderMailer < ApplicationMailer
+
+  def order(customer_email, cart, pickup_time)
```
Comment (tochman) [inline app/mailers/order_mailer.rb]: Not sure if I think this method name is suitable. Are there any conventions?

File: app/views/order_mailer/welcome_email.text.erb (hunk 3 above)
Comment (magnus-thor) [reply app/views/order_mailer/welcome_email.text.erb]: this was just a reminder on what we needed and will be removed soon.

File: spec/mailers/previews/order_mailer_preview.rb (hunk 2 above)
Comment (magnus-thor) [reply spec/mailers/previews/order_mailer_preview.rb:3]: came with generation of OrderMailer and it is to preview  the email, not sure how to use  it.

File: app/mailers/order_mailer.rb (hunk 6 above)
Comment (magnus-thor) [reply app/mailers/order_mailer.rb]: None that I found.

File: app/views/order_mailer/order.html.erb (hunk 4 above)
Comment (nomad-alt) [reply app/views/order_mailer/order.html.erb]: Can we add product_id as an order number into mailer?

Comment (magnus-thor) [reply app/views/order_mailer/order.html.erb]: If we did that then the people with same items in their order would have same order number

**Review by Sajaas — APPROVED** [review]
Good job!

**Pull request merged**